internal/ibmcloud-janitor/resources/vpc_lbs.go

```diff
@@ -0,0 +1,118 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"strings"
+	"time"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type VPCLoadBalancer struct{}
+
+var (
+	lbTimeout      = time.Minute * 10
+	lbWaitTime     = time.Minute * 1
+	resourceLogger *logrus.Entry
+)
+
+// Cleans up the load balancers in a given region
+func (VPCLoadBalancer) cleanup(options *CleanupOptions) error {
+	resourceLogger = logrus.WithFields(logrus.Fields{"resource": options.Resource.Name})
+	resourceLogger.Info("Cleaning up the load balancers")
+	client, err := NewVPCClient(options)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create VPC client")
+	}
+
+	var deletedLBList = make(map[string]bool)
+
+	f := func(start string) (bool, string, error) {
+		listLbOpts := &vpcv1.ListLoadBalancersOptions{}
+		if start != "" {
+			listLbOpts.Start = &start
+		}
+
+		loadBalancers, _, err := client.ListLoadBalancers(listLbOpts)
+		if err != nil {
+			return false, "", errors.Wrap(err, "failed to list the load balancers")
+		}
+
+		if loadBalancers == nil || len(loadBalancers.LoadBalancers) <= 0 {
+			resourceLogger.Info("there are no available load balancers to delete")
+			return true, "", nil
+		}
+
+		for _, lb := range loadBalancers.LoadBalancers {
+			if *lb.ResourceGroup.ID == client.ResourceGroupID {
+				if _, err := client.DeleteLoadBalancer(&vpcv1.DeleteLoadBalancerOptions{
+					ID: lb.ID,
+				}); err != nil {
+					return false, "", errors.Wrapf(err, "failed to delete the load balancer %q", *lb.Name)
+				}
+				deletedLBList[*lb.ID] = false
+				resourceLogger.WithField("name", *lb.Name).Info("load balancer deletetion triggered")
+			}
+		}
+
+		if loadBalancers.Next != nil && *loadBalancers.Next.Href != "" {
+			return false, *loadBalancers.Next.Href, nil
+		}
+
+		return true, "", nil
+	}
+
+	if err = pagingHelper(f); err != nil {
+		return errors.Wrapf(err, "failed to delete the load balancer")
+	}
+
+	// check if the LBs are properly deleted
+	if err := checkLBs(deletedLBList, client); err != nil {
+		return errors.Wrapf(err, "failed to check the deletion of load balancers")
+	}
+
+	resourceLogger.Info("Successfully deleted the load balancers")
+	return nil
+}
+
+func checkLBs(list map[string]bool, client *IBMVPCClient) error {
+	timeout := 0 * time.Minute
+	for timeout < lbTimeout {
+		for id, lbDeleted := range list {
+			lbID := id
+			if !lbDeleted {
+				lb, _, err := client.GetLoadBalancer(&vpcv1.GetLoadBalancerOptions{ID: &lbID})
+				if err != nil && strings.Contains(err.Error(), "cannot be found") {
+					list[id] = true
+					continue
+				} else if *lb.ProvisioningStatus == "delete_pending" {
+					break
+				} else {
+					return err
+				}
+			}
+		}
+		resourceLogger.Info("load balancer deletion is pending, checking again in a minute")
```

Comment:
Need to have a check here for to validate the LBs are processed or not before sleeping. 
Even though all the LB's deleted, it would sleep and loop till timeout becomes greater than lbTimeout.
Need to refactor this logic.

It's better if we can add some unit test cases.

Reply:
@Amulyam24 We can use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait@v0.25.2#PollImmediate this func to query something on a interval with a timeout.

I think this suits our requirement. Can you please take a look at this?

e.g usage https://github.com/openshift/hypershift/blob/main/cmd/infra/powervs/create.go#L727

Reply:
@dharaneeshvrd, made changes as per your suggestion. Tested it locally too, and it works fine. The LBs deletion took a little less than 2 minutes. I have added a polling interval of 30 secs and a timeout as 4 minutes. PTAL.

Reply:
Ok the new logic looks good to me.